openapi.yaml

```diff
@@ -8,6 +8,79 @@ info:
     email: mail@kopfsachen.org
     url: 'https://www.kopfsachen.org'
 paths:
+  '/self-service/registration/api':
+    get:
+      summary: Initate registration flow
+      tags:
+        - authentication
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/registration'
+        '500':
+          description: Internal Server Error
+  '/self-service/registration':
+    post:
+      parameters:
+        - $ref: '#/components/parameters/flow'
+      summary: Receive accountKey during registration
+      tags:
+        - authentication
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/registration-with-flow-id-response'
+        '500':
+          description: Internal Service Error
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/registration-with-flow-id-request'
+  '/self-service/login/api':
+    get:
+      tags:
+        - authentication
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/registration'
+        '500':
+          description: Internal Service Error
+  '/self-service/login/flow':
+    post:
+      parameters:
+        - $ref: '#/components/parameters/flow'
+      tags:
+        - authentication
+      responses:
+        '200':
+          description: OK
+        '500':
+          description: Internal Service Error
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/registration-with-flow-id-request'
```

Comment:
You're correct, just a quick note: The password of each user is by design the MD5 hash of this user's accountKey. As of now, you need to supply the password in the request as the according hash value (i.e. hash it in the frontend). We will eventually move this hashing to the backend (so that you just need to supply the accountKey without a password value), we just cannot provide timetable yet as to when this change is going to be in place.